### PR TITLE
[talk] full implementation

### PR DESCRIPTION
**Pull request by Wandalen**

### CODE REVIEW

File: users/Cargo.toml [hunk 1]
```diff
@@ -13,6 +13,8 @@ gettext-rs.workspace = true
 libc.workspace = true
 chrono.workspace = true
 syslog = "6.1"
+thiserror = "1.0.63"
+binrw = "0.14.0"
```
Comment (jgarzik) [inline users/Cargo.toml]: remove micro version in deps.  e.g. "1.0" rather than "1.0.63"

File: users/talk.rs [hunk 2]
```diff
@@ -0,0 +1,1744 @@
...
+//
+// This file is part of the posixutils-rs project covered under
+// the MIT License.  For the full license text, please see the LICENSE
+// file in the root directory of this project.
+// SPDX-License-Identifier: MIT
+//
+
+extern crate clap;
+extern crate gettextrs;
+extern crate plib;
```
Comment (jgarzik) [inline users/talk.rs]: extern crate no longer needed.

File: users/talk.rs [hunk 3]
```diff
@@ -0,0 +1,1744 @@
...
+            my_machine_addr,
+            &mut logger,
+            &mut output_buffer,
+        )?;
+    }
+
+    Ok(())
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
```
Comment (jgarzik) [inline users/talk.rs]: `main` should always be the last function in the program.

**Review by jgarzik — CHANGES_REQUESTED** [review]
Well done on the dependencies.  Only minor comments for now.

**Comment by Wandalen** [comment]
@jgarzik done

**Pull request merged**